### ISSUE TITLE
Querqy with Symspell rewritter

### ISSUE THREAD

**Issue by MighTguY** [querqy/querqy#120]
Integrating Querqy with Symspell, by writing a custom rewriter which can handle the spell correction 
1. Single token spell correction
2. Word break based spell correction

CC: @renekrie

**Comment by renekrie** [comment]
Thank you, @MighTguY!

If you want to go ahead with the implementation: would this introduce additional library dependencies? - I'd like to keep them at a minimum to avoid conflicts when Querqy is plugged into the search engines.

I assume that the Symspell rewriter wouldn't have any dependency on Lucene so that it could live in query-core?

**Comment by MighTguY** [comment]
Yes, I will add the [customized-symspell](https://github.com/MighTguY/customized-symspell) as a dependency, 
yes it wouldn't have any dependency over Lucene,

**Comment by renekrie** [comment]
Thank you. I think we'll have to find a way to deal with these dependencies as customized-symspell depends on a number of libraries. For example, I think we might run into version conflicts with apache-commons.* when Querqy is deployed to Solr/ES (which can probably be resolved). Also, as far as I know, ES doesn't use slf4j at all.
